### PR TITLE
[IMM32][NTUSER] Use HandleToUlong, UlongToHandle etc. macros

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Respect coding standard.
JIRA issue: N/A

## Proposed changes

- Use `HandleToUlong`, `UlongToHandle`, `UlongToPtr` etc. macros for type casting.
- Remove redundant casts.

### CODE REVIEW

File: dll/win32/imm32/ime.c [hunk 1]
```diff
@@ -297,7 +297,7 @@ PIMEDPI APIENTRY Imm32LoadImeDpi(HKL hKL, BOOL bLock)
     pImeDpiNew->hKL = hKL;
 
     lcid = LOWORD(hKL);
-    if (TranslateCharsetInfo((LPDWORD)(DWORD_PTR)lcid, &ci, TCI_SRCLOCALE))
+    if (TranslateCharsetInfo((LPDWORD)UlongToPtr(lcid), &ci, TCI_SRCLOCALE))
```
Comment (HBelusca) [inline dll/win32/imm32/ime.c:300]: The LPDWORD cast might be removed, if this doesn't cause MSVC nor GCC to warn or err.

File: dll/win32/imm32/imm.c [hunk 2]
```diff
@@ -171,7 +171,7 @@ BOOL WINAPI ImmFreeLayout(DWORD dwUnknown)
     }
     else
     {
-        hNewKL = (HKL)(DWORD_PTR)dwUnknown;
+        hNewKL = (HKL)UlongToHandle(dwUnknown);
```
Comment (HBelusca) [inline dll/win32/imm32/imm.c:174]: Is the HKL cast still needed? (here and in the other changes below and in the other files)

File: dll/win32/imm32/ime.c (hunk 1 above)
Comment (katahiromz) [reply dll/win32/imm32/ime.c]: Removed.

File: dll/win32/imm32/imm.c (hunk 2 above)
Comment (katahiromz) [reply dll/win32/imm32/imm.c]: Deleted.

**Pull request merged**